backend/pkg/kubeconfig/kubeconfig.go

```diff
@@ -258,357 +279,492 @@ func (c *Context) AuthType() string {
-			}
+	userName := contextData["user"].(string)
 
-			err := SetAuthInfoField(authInfo, strKey, value)
-			if err != nil {
-				return nil, UserError{UserName: userName, Reason: err.Error()}
-			}
-		}
+	return clusterName, userName, nil
+}
+
+// getClusterAndUser gets the cluster and user details from the kubeconfig.
+func getClusterAndUser(
+	kubeconfig map[string]interface{},
+	clusterName,
+	userName string,
+) (map[interface{}]interface{}, map[interface{}]interface{}, error) {
+	var errs []error
+
+	cluster, err := getCluster(kubeconfig, clusterName)
+	if err != nil {
+		errs = append(errs, err)
+	}
 
-		return authInfo, nil
+	user, err := getUser(kubeconfig, userName)
+	if err != nil {
+		errs = append(errs, err)
 	}
 
-	return nil, UserError{UserName: userName, Reason: "user not found"}
+	return cluster, user, errors.Join(errs...)
 }
 
-// createAuthProvider creates an auth provider from the given value.
-// It returns the created auth provider and any errors that occurred.
-// It returns an error if the auth provider cannot be created.
-// It returns a valid auth provider if the auth provider is created.
-// It returns any errors that occurred during auth provider creation.
-func createAuthProvider(value interface{}) *api.AuthProviderConfig {
-	provider, ok := value.(map[interface{}]interface{})
-	if !ok {
-		return nil
+// createAndValidateConfig creates and validates the config.
+func createAndValidateConfig(
+	contextName string,
+	contextMap,
+	cluster,
+	user map[interface{}]interface{},
+	kubeconfig map[string]interface{},
+) (*api.Config, error) {
+	singleConfig := createKubeConfig(contextName,
+		toStringKeyMap(contextMap),
+		toStringKeyMap(cluster),
+		toStringKeyMap(user))
+
+	yamlData, err := yaml.Marshal(singleConfig)
+	if err != nil {
+		return nil, ContextError{
+			ContextName: contextName,
+			Reason:      fmt.Sprintf("error marshaling to YAML: %v", err),
+		}
 	}
 
-	authProvider := &api.AuthProviderConfig{}
-	authProvider.Name, _ = provider["name"].(string)
+	clientConfig, err := clientcmd.Load(yamlData)
+	if err != nil {
+		clusterName := getNameOrUnknown(cluster, "name")
+		userName := getNameOrUnknown(user, "name")
+
+		return nil, HandleConfigLoadError(err, contextName, clusterName, userName, kubeconfig)
+	}
 
-	if config, ok := provider["config"].(map[interface{}]interface{}); ok {
-		authProvider.Config = make(map[string]string)
+	return clientConfig, nil
+}
 
-		for k, v := range config {
-			if key, ok := k.(string); ok {
-				if val, ok := v.(string); ok {
-					authProvider.Config[key] = val
-				}
-			}
+// getNameOrUnknown returns the name of the cluster or user if it exists, otherwise it returns "<unknown>".
+func getNameOrUnknown(data map[interface{}]interface{}, key string) string {
+	if nameInterface, ok := data[key]; ok {
+		if nameString, ok := nameInterface.(string); ok {
+			return nameString
 		}
 	}
 
-	return authProvider
+	return "<unknown>"
```

Comment:
Maybe just "unknown"? I don't know if "<>" are valid. 

Or for context name could possible use (cluster name and user name?)